pipeline/transformer/transformer_calm/src/main/scala/uk/ac/wellcome/platform/transformer/calm/TransformerWorker.scala

```diff
@@ -37,34 +37,42 @@ trait TransformerWorker[In, SenderDest] extends Logging {
   type Result[T] = Either[TransformerWorkerError, T]
   type StoreKey = Version[String, Int]
 
+  sealed trait TransformResult;
+  case class Success(key: StoreKey, work: TransformedBaseWork) extends TransformResult
+  case class Surpress(key: StoreKey) extends TransformResult
+  case class Error(error: TransformerWorkerError) extends TransformResult
+
   def name: String = this.getClass.getSimpleName
   val stream: SQSStream[NotificationMessage]
   val sender: BigMessageSender[SenderDest, TransformedBaseWork]
   val store: VersionedStore[String, Int, In]
   val transformer: Transformer[In]
   val concurrentTransformations: Int = 2
 
-  def process(
-    message: NotificationMessage): Result[(TransformedBaseWork, StoreKey)] = {
-    for {
+  def process(message: NotificationMessage): TransformResult =
+    (for {
       key <- decodeKey(message)
-      recordAndKey <- getRecord(key)
-      work <- work(recordAndKey._1, key)
-      done <- done(work, key)
-    } yield done
-  }
+      record <- getRecord(key)
+      shouldTransform = transformer.shouldTransform(record)
+      result <- if (!shouldTransform) Right(Surpress(key)) else for {
+        success <- work(record, key)
+        _ <- done(success)
+      } yield success
```

Comment:
This is just a bit much on one line for my taste. Could you break out the inner for comprehension to another function?